### PR TITLE
Update components to handle terra-list v4

### LINKED ISSUE #418: [terra-popup] Remove unneeded div when open, use react fragment
# Bug Report

## Description
Remove the unneeded div that wraps the portal for the popup and overlay. This can be converted to a React.Fragment to avoid unintentional layout issues that a blank element can add.

## Steps to Reproduce
1. Open popup, observe empty div added to DOM.

### LINKED ISSUE #428: popUp wiki entry contains broken links
# Bug Report

## Description

On the following page 
https://engineering.cerner.com/terra-ui/#/components/terra-popup/popup/popup
there is a section that contains links to an App delegate page. However clicking on said links results in a 404 error 

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-ui/#/components/terra-popup/popup/popup
2. find the Rendering a popup inside a modal section
3. click on the blue hyperlinked text "App delegate"

## Expected Behavior

the link should direct to a page describing the app deligate

## Possible Solution

update the embedded hyperlink

## Environment

* Browser Name and Version: Chrome
* Operating System and version (desktop or mobile): Mac OS

### LINKED ISSUE #438: [terra-popup] Update modal documentation
# Feature Request
Popup has a documentation section called `Rendering a popup inside a modal` that uses the AppDelegate pattern. AppDelegate has been replaced with DisclosureManager's new context API - let's update the documentation to reflect this.

### LINKED ISSUE #511: Minified React Error Seen at engineering.cerner.com/terra-core
# Bug Report

## Description

Saw 3 occurrences of this error:

Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.

I'd like for us to do a little digging to try and figure out why this happened.

### PR DESCRIPTION
**Pull request by dkasper-was-taken**
### Summary
Update to list v4

### Additional Details
Remove dependency on terra-list if possible, otherwise update major version bumps for package.

Resolves #418 
Resolves #428 
Resolves #438 
Resolves #511 

Thanks for contributing to Terra.
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-popup/src/Popup.jsx [hunk 1]
```diff
@@ -286,7 +286,7 @@ class Popup extends React.Component {
     const hookshotContent = this.createPopupContent(boundingRef ? boundingRef() : undefined, showArrow);
 
     return (
-      <div>
+      <React.Fragment>
```
Comment (bjankord) [inline packages/terra-popup/src/Popup.jsx]: This resolves https://github.com/cerner/terra-framework/issues/418 right?

File: packages/terra-collapsible-menu-view/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,9 @@ ChangeLog
 
 Unreleased
 ----------
+### Breaking Change
+* Update to terra-list v4, the result of themeable variables changing
```
Comment (emilyrohrbough) [inline packages/terra-collapsible-menu-view/CHANGELOG.md]: Do we need to include the to the upgrade guide?

File: packages/terra-collapsible-menu-view/src/CollapsibleMenuViewItemGroup.jsx [hunk 3]
```diff
@@ -80,7 +80,7 @@ class CollapsibleMenuViewItemGroup extends React.Component {
     } if (isCollapsibleMenuItem) {
       return (
         <li>
-          <List {...customProps} onChange={this.handleOnChange}>
+          <List {...customProps} role="listbox">
```
Comment (emilyrohrbough) [inline packages/terra-collapsible-menu-view/src/CollapsibleMenuViewItemGroup.jsx]: Did we need to pass the `onChange` still?

File: packages/terra-infinite-list/CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,17 @@ Changelog
...
 ----------
+### Breaking Change
+* Update to terra-list v4
+* -- Update props to match List's new API
+* -- Selection state now managed outside of the infinite list
+* -- Expose SectionHeader and SubsectionHeader for use within list
+* -- Removed isDivided prop
+* -- Removed isSelectable prop
+* -- Removed selectedIndexes prop
+* -- Removed hasChevrons prop
```
Comment (emilyrohrbough) [inline packages/terra-infinite-list/CHANGELOG.md]: Looks like disableUnselectedItems was also removed

File: packages/terra-infinite-list/CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,17 @@ Changelog
 
 Unreleased
 ----------
+### Breaking Change
+* Update to terra-list v4
+* -- Update props to match List's new API
+* -- Selection state now managed outside of the infinite list
+* -- Expose SectionHeader and SubsectionHeader for use within list
+* -- Removed isDivided prop
```
Comment (emilyrohrbough) [inline packages/terra-infinite-list/CHANGELOG.md]: - Replaced `isDivided` prop with `dividerStyle`?
-

File: packages/terra-infinite-list/src/InfiniteList.jsx [hunk 6]
```diff
@@ -3,71 +3,58 @@ import PropTypes from 'prop-types';
...
   /**
    * An indicator to be displayed at the end of the current loaded children.
    */
   progressiveLoadingIndicator: PropTypes.element,
   /**
-   * An array of the currectly selected indexes.
+   * Accessibility role of the list, defaults to 'none'. If creating a list with selectable items, pass 'listbox'.
    */
-  selectedIndexes: PropTypes.array,
+  role: PropTypes.string,
```
Comment (emilyrohrbough) [inline packages/terra-infinite-list/src/InfiniteList.jsx]: Should this be one of `none` or `listbox`?

File: packages/terra-infinite-list/src/InfiniteList.jsx [hunk 7]
```diff
@@ -519,6 +497,10 @@ class InfiniteList extends React.Component {
 
 InfiniteList.propTypes = propTypes;
 InfiniteList.defaultProps = defaultProps;
-InfiniteList.Item = List.Item;
 
 export default InfiniteList;
+export {
+  Item,
+  SectionHeader,
+  SubsectionHeader,
```
[the remark this answers is not in the document]
Comment (emilyrohrbough) [reply packages/terra-infinite-list/src/InfiniteList.jsx]: This is a MVB. Should we move towards the index.js format that exports each of these like we did with table? https://github.com/cerner/terra-core/blob/2cff28583a26c51f75479376c71532249a86b614/packages/terra-table/src/index.js Not sure it matters here though

File: packages/terra-menu/CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,8 @@ Changelog
 
 Unreleased
 ----------
+### Breaking Change
+* Update to terra-list v4, the result of themeable variables changing
```
Comment (emilyrohrbough) [inline packages/terra-menu/CHANGELOG.md]: Should we link an upgrade guide?

File: packages/terra-menu/tests/wdio/menu-spec.js [hunk 9]
```diff
@@ -81,7 +81,7 @@ describe('Menu', () => {
       browser.click('#selectable-menu-button');
       browser.hasFocus('[class*="content"][aria-modal="true"][role="dialog"]');
       browser.click('.TestGroupItem3');
-      browser.hasFocus('li:last-child[aria-selected="true"][role="menuitemradio"]');
+      browser.hasFocus('li:last-child[aria-checked="true"][role="menuitemradio"]');
```
Comment (emilyrohrbough) [inline packages/terra-menu/tests/wdio/menu-spec.js]: I'm unfamiliar, what caused the change from aria-selected to aria-checked? List's implementation?

File: packages/terra-navigation-side-menu/src/_MenuItem.jsx [hunk 10]
```diff
@@ -99,28 +98,29 @@ class MenuItem extends React.Component {
...
-            className={itemClassNames}
-            onKeyDown={this.handleKeyDown}
-            onKeyUp={this.handleKeyUp}
-            onBlur={this.handleOnBlur}
-          >
-            <div className={cx('title')}>
-              {text}
-            </div>
-            {hasChevron && <span className={cx('chevron')}><ChevronRight /></span>}
+        aria-selected="false"
```
Comment (emilyrohrbough) [inline packages/terra-navigation-side-menu/src/_MenuItem.jsx]: shouldn't this value toggle with the `isSelected` prop value?

File: packages/terra-collapsible-menu-view/src/CollapsibleMenuViewItemGroup.jsx (hunk 3 above)
Comment (dkasper-was-taken) [reply packages/terra-collapsible-menu-view/src/CollapsibleMenuViewItemGroup.jsx]: onChange doesn't do anything for List. I believe it was added by mistake, onChange only triggers from a selectable variant of list.

File: packages/terra-infinite-list/src/InfiniteList.jsx (hunk 6 above)
Comment (dkasper-was-taken) [reply packages/terra-infinite-list/src/InfiniteList.jsx]: There are quite a few variants, so we can't really lock it down.

File: packages/terra-menu/tests/wdio/menu-spec.js (hunk 9 above)
Comment (dkasper-was-taken) [reply packages/terra-menu/tests/wdio/menu-spec.js]: It was a failing test that we didn't catch until list broke navigation.

File: packages/terra-navigation-side-menu/src/_MenuItem.jsx (hunk 10 above)
Comment (dkasper-was-taken) [reply packages/terra-navigation-side-menu/src/_MenuItem.jsx]: Good catch, looks like this was a bug in the previous code as well.

File: packages/terra-infinite-list/src/InfiniteList.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (dkasper-was-taken) [reply packages/terra-infinite-list/src/InfiniteList.jsx]: I'll make the index.js change in this PR, but I logged an issue to create additional documentation:
https://github.com/cerner/terra-framework/issues/512

File: packages/terra-collapsible-menu-view/CHANGELOG.md (hunk 2 above)
Comment (dkasper-was-taken) [reply packages/terra-collapsible-menu-view/CHANGELOG.md]: Updated.

File: packages/terra-infinite-list/CHANGELOG.md (hunk 4 above)
Comment (dkasper-was-taken) [reply packages/terra-infinite-list/CHANGELOG.md]: Updated.

File: packages/terra-infinite-list/CHANGELOG.md (hunk 5 above)
Comment (dkasper-was-taken) [reply packages/terra-infinite-list/CHANGELOG.md]: Updated.

File: packages/terra-navigation-side-menu/src/_MenuItem.jsx (hunk 10 above)
Comment (dkasper-was-taken) [reply packages/terra-navigation-side-menu/src/_MenuItem.jsx]: Updated.

File: packages/terra-navigation-side-menu/src/_MenuItem.jsx [hunk 11]
```diff
@@ -102,7 +102,7 @@ class MenuItem extends React.Component {
     return (
       <li
         className={cx('list-item')}
-        aria-selected="false"
+        aria-selected={isSelected}
```
Comment (emilyrohrbough) [inline packages/terra-navigation-side-menu/src/_MenuItem.jsx]: does this get passed as a string value?

Comment (dkasper-was-taken) [reply packages/terra-navigation-side-menu/src/_MenuItem.jsx]: React converts it to a string either way, but it's a bool.

**Pull request merged**